### DISCUSSION TITLE
Annotations API should return dashboard UID

### DISCUSSION THREAD

**Discussion by djzurawski** [grafana/grafana#37028] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Instead of, or in addition to dashboardId, the annotations http api should return the dashboard UID

**Why is this needed**:
You cannot query dashboard info with dashboardID, only the UID. You cannot go to a dashboard url with the dashboardId. If I want to know what dashboard an annotation belongs to I have to manually go into every dashboard and look at the JSON model under settings.

**Comment by torkelo** [comment]
Anyone wants to help here is the place:
https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/annotation.go#L129

**Comment by cunyat** [comment]
Hey @torkelo, here would be enought to make a join in that query with dasboards, and add dasboardUid to itemDTO struct?
I was looking where i could place a test for that, but i really can't figure it out

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
